client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -210,29 +251,34 @@ class CalypsoifyIframe extends Component {
 		this.setState( { classicBlockEditorId: null, isMediaModalVisible: false } );
 	};
 
-	insertClassicBlockMedia = markup => {
-		if ( !! this.state.classicBlockEditorId && this.iframePort ) {
-			this.iframePort.postMessage( {
-				action: 'insertClassicBlockMedia',
-				payload: {
-					editorId: this.state.classicBlockEditorId,
-					media: markup,
-				},
-			} );
+	insertClassicBlockMedia = ( markup: string ) => {
+		if ( ! ( this.state.classicBlockEditorId && this.iframePort ) ) {
+			return;
 		}
+
+		this.iframePort.postMessage( {
+			action: 'insertClassicBlockMedia',
+			payload: {
+				editorId: this.state.classicBlockEditorId,
+				media: markup,
+			},
+		} );
 	};
 
 	pressThis = () => {
 		const { pressThis } = this.props;
-		if ( pressThis ) {
-			this.iframePort.postMessage( {
-				action: 'pressThis',
-				payload: pressThis,
-			} );
+
+		if ( ! ( pressThis && this.iframePort ) ) {
+			return;
 		}
+
+		this.iframePort.postMessage( {
+			action: 'pressThis',
+			payload: pressThis,
+		} );
 	};
 
-	updateImageBlocks = action => {
+	updateImageBlocks = ( action: { data: { mime_type: string; URL: string }; type: string } ) => {
```

Comment:
Having a separate typedef for this action would be more readable than the inline type.

Reply:
I'm not sure what to do in these cases. I guess we can create types or interfaces for them, but I like having these things inline so I don't have to jump around to find them. much of my trouble here has to do with the fact that I don't know what these should be so I'm going for minimum effort on typing. I'll think about ways to do this otherwise

Reply:
The Happy Tools folks use Flow to typecheck their Redux code. They have typedefs for actions that allow type inference by the `type` value:
```js
switch ( action.type ) {
  case 'UPDATE_IMAGE_BLOCKS':
    // at this place your editor knows what fields the `action` has
}
```
See p9lV3a-r7-p2. That could be a way to typecheck Redux with TypeScript, too.

Reply:
thanks @jsnajdr. it's true we have several options here to type on. I guess you are saying that this is a Redux action - I didn't realize that at the time.

typing actions is a broader question and one which will undoubtedly push into more advanced type annotations. it's entirely possible though.

I'll look into that action and see what I can come up with. in some cases I think I'd still fall back to providing a minimal type until we have more of the core modules typed. that is, getting our actions typed might be a huge project, but I can still let the callee know what I need and expect here with these few parameters.

we can do a funky mid-term solution here as well though I wouldn't personally introduce this.

```js
( action: { … }, type: 'UPDATE_IMAGE_BLOCKS' as const ) => { … }
```

what it brings us is a warning at the call site if we try and send any action of a different type. now, a benefit of using the `@babel/typescript` preset is that it won't refuse to compile if we have this type error there. `tsc` _would_ refuse until we could provide ample evidence that only that string literal can be passed in as `type`